src/components/SearchInput.js

```diff
@@ -62,6 +75,35 @@ const SearchInput = (props) => {
   const fullInputRef = React.createRef()
   const [anchorEl, setAnchorEl] = React.useState(null)
   const [currentSearchEngine, setCurrentSearchEngine] = useState(searchEngine)
+  const [
+    currentYahooPaidSearchRewardOptIn,
+    setCurrentYahooPaidSearchRewardOptIn,
+  ] = useState(yahooPaidSearchRewardOptIn)
+  const [tooltipOpen, setTooltipOpen] = useState(tooltip)
+
+  const getSearchEngine = useCallback(
+    (searchEngineId) =>
+      searchEngines.edges.find(
+        (engine) => engine.node.engineId === searchEngineId
+      ).node,
+    [searchEngines]
+  )
+
+  useEffect(() => {
+    setCurrentSearchEngine(searchEngine)
+    setCurrentYahooPaidSearchRewardOptIn(
+      setYahooPaidSearchRewardOptIn !== undefined
+        ? setYahooPaidSearchRewardOptIn
+        : yahooPaidSearchRewardOptIn
+    )
+    setTooltipOpen(tooltip)
+  }, [
+    searchEngine,
+    getSearchEngine,
+    setYahooPaidSearchRewardOptIn,
+    yahooPaidSearchRewardOptIn,
+    tooltip,
+  ])
```

Comment:
Here, you're making some state from props, which isn't a great design pattern. Now you need to worry about both, and they might become misaligned in weird ways.

Instead, rely on Relay. In your mutations where data changes, include all potentially-changed fields in the mutation fragment. When the mutation fires, it'll automatically update Relay containers with their new values. In this case, the mutations that are missing fetched data are probably SetYahooSearchOptInMutation and SetUserSearchEngineMutation.

The caveat for this repo is that our data flow has a bug for data on the top-level pages. Changed Relay data does not currently update on the page level, only for data used within Relay containers. We'll eventually fix that but have used component state as a workaround. That bug shouldn't affect you here, because this component is in a Relay container.

Reply:
The updateImpact mutation is a reasonable example of fetching changed data:
https://github.com/gladly-team/tab-web/blob/ff305496255cf41cd4cc9609e1cc2174e77b5f5b/src/utils/mutations/UpdateImpactMutation.js#L8